pkg/cache/task.go

```diff
@@ -238,6 +242,21 @@ func (task *Task) postTaskAllocated(event *fsm.Event) {
 		// task allocation UID is assigned once we get allocation decision from scheduler core
 		task.allocationUuid = allocUuid
 
+		// before binding pod to node, first bind volumes to pod
+		log.Logger.Debug("bind pod volumes",
+			zap.String("podName", task.pod.Name),
+			zap.String("podUID", string(task.pod.UID)))
+		if task.volumeBinder != nil {
+			if err := task.volumeBinder.Binder.BindPodVolumes(task.pod); err != nil {
```

Comment:
I think we should use cached pod instead of task pod here, since task pod didn't update its Spec.NodeName field and this call will fail when task.pod.Spec.NodeName is "".
I have tested successfully for pods with PVC but failed for pods without PVC, and I can't find where task.pod is updated for pods with PVC, can you pls figure out that?

Reply:
I am still looking at this. I think there might be some inconsistency in the cache somewhere, but could not figure out yet before my laptop battery dead... will update on this tomorrow.

Reply:
I've updated a new commit. Hopefully, it can resolve this issue.
I used a workaround by calling AssumePodVolumes again right before BindPodVolumes, this is to get up-to-date volume bindings info in the volume-binder's cache, I am not sure if there is any other better approach to fix this.
@TaoYang526 could you please try this and let me know your thought? Thx

Reply:
Thanks @yangwwei for this update. I'm not sure if we need to call AssumePodVolumes again, I thought call BindPodVolumes with assumed pod should be fine. I'll try this again quickly then feedback a few minutes later.

Reply:
@yangwwei  I have tested this PR, it works fine after fixing a small issue as follows.
```
-               allBound, err := ctx.volumeBinder.Binder.AssumePodVolumes(assumedPod, pod.Spec.NodeName)
+               allBound, err := ctx.volumeBinder.Binder.AssumePodVolumes(assumedPod, assumedPod.Spec.NodeName)
```
I have also tested directly call BindPodVolumes without check if all volumes have been bound, it will fail for pods without unbound volume. So it's necessary to call AssumePodVolumes again and make sure not all volumes are bound before calling BindPodVolumes.

Reply:
Thanks for the quick update! Just updated the code to fix this.